float64[] local_pos
